moveit_core/robot_model/include/moveit/robot_model/link_model.h

```diff
@@ -41,13 +41,16 @@
 #include <vector>
 #include <utility>
 #include <map>
+#include <boost/shared_ptr.hpp>
 #include <Eigen/Geometry>
 #include <eigen_stl_containers/eigen_stl_vector_container.h>
 #include <moveit/macros/class_forward.h>
 
 namespace shapes
 {
-MOVEIT_CLASS_FORWARD(Shape);
+class Shape;
+typedef boost::shared_ptr<Shape> ShapePtr;
+typedef boost::shared_ptr<const Shape> ShapeConstPtr;
```

Comment:
Ah, these forward delarations where added by @rhaschke in 13d2e622658c995e140f85a08891ea65ba8d7dd5 to reduce includes of geometric_shapes.
I don't like the idea to introduce a header there that only forward declares all the types.
@rhaschke is this change really worth the overhead to maintain the boost-shared_ptrs here?

On the other hand I agree with @davetcoleman. I don't think it's any good to have boost- *and* std- shared_ptr around in `geometric_shapes` in kinetic. So I would actually propose to fully move to `std::shared_ptr` there. This would also simplify the forward declarations here.